File: src/main/java/com/elissamc/ElissaMC.java
package com.elissamc;

import cn.nukkit.event.Listener;
import cn.nukkit.plugin.PluginBase;
import cn.nukkit.utils.TextFormat;
import com.elissamc.instance.Instance;

import java.io.File;

public class ElissaMC extends PluginBase implements Listener {
    private Instance instance;
    public static File dataFolder;
    public static ElissaMC plugin;


    @Override
    public void onEnable() {
        getLogger().info(TextFormat.RED + "ElissaMC has been enabled!");
        plugin = this;
        dataFolder = this.getDataFolder();
        getServer().getPluginManager().registerEvents( instance = new Instance(this), this);
        instance.load();
    }

    public Instance getInstance() {
        return instance;
    }

}

File: src/main/kotlin/com/elissamc/ElissaMC.kt
package com.elissamc

import cn.nukkit.Server
import cn.nukkit.event.Listener
import cn.nukkit.level.Level
import cn.nukkit.plugin.PluginBase
import cn.nukkit.utils.TextFormat
import java.io.File

class ElissaMC : PluginBase(), Listener {

    override fun onEnable() {
        logger.info(TextFormat.RED.toString() + "ElissaMC has been enabled!")
        this.init()
    }

    fun init() {
        ElissaMC.plugin = this
        ElissaMC.folder = this.dataFolder
        ElissaMC.core = ElissaCore(this)
        server.pluginManager.registerEvents(core, this)
    }

    companion object {
        @JvmStatic
        fun getWorld(world: String): Level? {
            return Server.getInstance().getLevelByName(world)
        }

        lateinit var plugin: ElissaMC
        lateinit var folder: File
        lateinit var core: ElissaCore
    }

}
